securedrop/tests/test_2fa.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+import os
+import random
+
+from flask import url_for
+from flask_testing import TestCase
+
+os.environ['SECUREDROP_ENV'] = 'test'
+from db import (db_session, BadTokenException)
+import journalist
+import utils
+
+# Smugly seed the RNG for deterministic testing
+random.seed('¯\_(ツ)_/¯')
+
+
+class TestJournalist2FA(TestCase):
+    def create_app(self):
+        return journalist.app
+
+    def setUp(self):
+        utils.env.setup()
+
+        self.admin, self.admin_pw = utils.db_helper.init_journalist(
+            is_admin=True)
+
+    def tearDown(self):
+        utils.env.teardown()
+        # TODO: figure out why this is necessary here, but unnecessary in all
+        # of the tests in `tests/test_unit_*.py`. Without this, the session
+        # continues to return values even if the underlying database is deleted
+        # (as in `shared_teardown`).
+        db_session.remove()
```

Comment:
That is weird since the first line of the last function is this one. It's not weird that you need to call this method on `db_session` despite the database being deleted (that part makes sense), but that calling it the first time doesn't work.